astropy/table/column.py

```diff
@@ -407,6 +410,42 @@ def name(self, val):
         self._name = val
 
     @property
+    def format(self):
+        """
+        Format string for displaying values in this column.
+        """
+
+        return self._format
+
+    @format.setter
+    def format(self, format_string):
+        if format_string is None:
+            self._format = None
+            return
+
+        try:
+            prev_format = self._format  # save current format string
+        except AttributeError:  # nothing set at all yet
+            prev_format = None
+
+        self._format = format_string  # set new format string
+
+        try:
+            # test whether it formats without error exemplarily
+            self.pformat(max_lines=1)
+
+            if self.dtype.kind == 'O':
```

Comment:
Even though this is formally correct to do the full pre-checking here, I think it is better to not have a very expensive operation.  One could end up formatting a billion lines just to set the`format` attribute, which is not a good thing.